uiowa/uiowa
# BLT UMD - Application check added to database operation

Resolves https://github.com/uiowa/uiowa/issues/7996

The main change with this is that it gets the $app variable early in the process from the drush alias info and uses that for the files deletion. When the file deletion code was added, the database portion of the command was not updated to leverage this info.

The command is set to return early if the app does not exist.

So with the $app variable, instead of looping through all of the apps, it targets that specific app and returns back just the databases that reside there. Since there can't be duplicate database names on the same app the rest of the command works as expected.

# How to test

- See that https://github.com/uiowa/uiowa/pull/8013/commits/8262348cdd038768185f79550d231a5b471b0a16 was successful in deleting the gel_sites_uiowa_edu db from uiowa05
- Provision two test sites foo.lab.uiowa.edu (e.g. uiowa) and foo-lab.uiowa.edu (e.g. uiowa07). Run the provision command with the --no-commit flag to create the databases on the remotes. Now run the umd command, selecting foo-lab.uiowa.edu and making sure that the uiowa07 databases is removed and not the uiowa databases. Follow up with the deprovision of foo.lab and see that it is also successful. reset and discard git commits from the deprovisions.
